torch/csrc/dynamo/eval_frame.c

```diff
@@ -320,6 +320,53 @@ inline static CacheEntry* get_cache_entry(PyCodeObject* code) {
   return extra;
 }
 
+PyObject* _debug_get_cache_entry_list(PyObject* self, PyObject* args) {
+  PyObject* my_object;
+  if (!PyArg_ParseTuple(args, "O", &my_object)) {
+    return NULL;
+  }
+  PyCodeObject* code = (PyCodeObject*)my_object;
+
+  CacheEntry* current_node = get_cache_entry(code);
+
+  PyObject* outer_list = PyList_New(0);
+  if (!outer_list) {
+    return NULL;  // Return NULL if failed to create list
+  }
+  while (current_node != NULL) {
+    // Creating a new Python list for the check_fn and code of current CacheEntry
+    PyObject* inner_list = PyList_New(0);
```

Comment:
I believe you a leaking a reference to this object.

Reply:
Shouldn't be a problem now, since we are using `PyTuple_Pack`.